### ISSUE TITLE
Package request: icloud-for-linux

### ISSUE THREAD

**Issue by hartman8227** [NixOS/nixpkgs#270060]
Project enables syncing with iCloud on Linux. Presently is seems to only be packaged in snap format.

**Metadata**

* homepage URL: https://github.com/cross-platform/icloud-for-linux
* source URL: https://github.com/cross-platform/icloud-for-linux
* license: GPL 3
* platforms: Linux

**Priorities**

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by 2gn** [comment]
I noticed that the description of the software is quite misleading. It doesn't allow "synching with iCloud" and most likely it's just a wrapper for icloud.com.

**Comment by hartman8227** [comment]
Sorry if I've misrepresented what it is, but that is the effect. How it does this is beyond my current knowledge of computers. I drive trucks for a living.

If it's not possible to package this for nixos, any ideas on how I might get it running as is? Snapd also seems not to be currently available for Nixos. At least I haven't been able to find it yet. 

Or should I give up running this on my laptop and do a workaround from my Ubuntu based NextCloud server?

**Comment by 2gn** [comment]
It is technically possible to package the software. However, you can access the same functionality provided by the software by visiting https://icloud.com.

**Comment by hartman8227** [comment]
If I recall correctly, bearing in mind I last daily drove Linux 2ish years ago, this package did provide what amounted to web shortcuts to various iCloud services in the dock applet, however it also provided a mechanism to sync a directory in my home folder with a directory in my iCloud storage. 

How? No clue. But that is the functionality I need.

When I get back home from my route I have a reset and I'll spin up an Ubuntu install in a VM and make sure this is the package I think it is. Should be as it is still the only thing I can find in Linux land that has anything to do with iCloud.

**Comment by 2gn** [comment]
If it's real, I have no idea how it works. It's either my lack of knowledge, or the lack of important parts missing from the code on the repository.

**Comment by hartman8227** [comment]
So either my memory is crap or this package has changed as it can't do syncing anymore. I'm closing this request as I don't need a bunch of links.

If anyone in the future is trying to sync iCloud to linux, the workaround I've set up is to use a Windows box to sync iCloud and then syncing the same folder to Nextcloud. Crappy way round, but it's the best idea I've got right now. Mac to Nextcloud ought to work as well.
